core/src/main/java/com/lafaspot/sparzr/InternalSaxHandler.java

```diff
@@ -0,0 +1,355 @@
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+/**
+ * Encompasses all SAX logic required for parsing schema.org annotations. Currently supports Json-ld and Microdata formats with all data normalized to
+ * json.
+ *
+ * Microdata to json conversion based on <a href="http://www.w3.org/TR/microdata/#json">W3 spec</a>
+ *
+ * @author akajla
+ *
+ */
+public class InternalSaxHandler extends DefaultHandler {
+
+    /**
+     * Script type defining json-ld format.
+     */
+    private static final String JSONLD_SCRIPT_TYPE = "application/ld+json";
+
+    /**
+     * script tag.
+     */
+    private static final String SCRIPT_TAG = "script";
+
+    /**
+     * Microdata itemscope attribute.
+     */
+    private static final String ITEMSCOPE_ATTRIBUTE = "itemscope";
+
+    /**
+     * Microdata itemtype attribute.
+     */
+    private static final String ITEMTYPE_ATTRIBUTE = "itemtype";
+
+    /**
+     * Microdata itemprop attribute.
+     */
+    private static final String ITEMPROP_ATTRIBUTE = "itemprop";
+
+    /**
+     * Generic "type" attribute.
+     */
+    private static final String TYPE_ATTRIBUTE = "type";
+
+    /**
+     * List of references to items for traversing item hierarchy.
+     */
+    private final LinkedList<ObjectNode> items;
+
+    /**
+     * Stack to keep track of item state for every tag.
+     */
+    private final Stack<Format> tagTypes;
+
+    /**
+     * Currently registered list of listeners called during parsing.
+     */
+    private final List<Listener> listeners;
+
+    /**
+     * Indicate whether current tag contains Json-ld data.
+     */
+    private boolean containsJsonld;
+
+    /**
+     * Total items processed by parser during run.
+     */
+    private int totalItemtypes;
+
+    /**
+     * Buffer to use for reading string data in between tags.
+     */
+    private StringBuffer stringBuffer;
+
+    /**
+     * Current item property being parsed.
+     */
+    private String propertyName;
+
+
+    /**
+     * JsonNode factory to be used to create respective types of nodes.
+     */
+    private final JsonNodeFactory jsonNodeFactory;
+
+    /**
+     * The mapper to convert string to JSON and vice versa.
+     */
+    private final ObjectMapper mapper;
+
+    /**
+     * Create an object that will use provided listeners to notify caller of events.
+     *
+     * @param listeners
+     *            Client provided list of listeners to be called on parser events
+     */
+    public InternalSaxHandler(final List<Listener> listeners) {
```

Comment:
missing notnull  or nullable annotattion, can you add that to the checkstyle rules.